packages/app-utils/src/components/ConnectionBootstrap.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { createContext, useEffect, useState } from 'react';
+import { LoadingOverlay } from '@deephaven/components';
+import { useApi } from '@deephaven/jsapi-bootstrap';
+import { IdeConnection } from '@deephaven/jsapi-types';
+import Log from '@deephaven/log';
+import useClient from './useClient';
+
+const log = Log.module('@deephaven/jsapi-components.ConnectionBootstrap');
+
+export const ConnectionContext = createContext<IdeConnection | null>(null);
+
+export type ConnectionBootstrapProps = {
+  /**
+   * The children to render wrapped with the ConnectionContext.
+   * Will not render children until the connection is created.
+   */
+  children: React.ReactNode;
+};
+
+/**
+ * ConnectionBootstrap component. Handles initializing the connection.
+ */
+export function ConnectionBootstrap({ children }: ConnectionBootstrapProps) {
+  const api = useApi();
+  const client = useClient();
+  const [error, setError] = useState<unknown>();
+  const [connection, setConnection] = useState<IdeConnection>();
+  useEffect(
+    function initConnection() {
+      let isCancelled = false;
+      async function loadConnection() {
+        try {
+          const newConnection = await client.getAsIdeConnection();
+          if (isCancelled) {
+            return;
+          }
+          setConnection(newConnection);
+        } catch (e) {
+          if (isCancelled) {
+            return;
+          }
+          setError(e);
+        }
+      }
+      loadConnection();
+      return () => {
+        isCancelled = true;
+      };
```

Comment:
Oh yea... logging out. We'd need to just create a whole new client, so should probably be handled in `ClientBootstrap`. And listen to a broadcast message (since we want to log out all tabs at the same time).

Reply:
Make a followup ticket for this?

Reply:
#1239